### PR TITLE
Workflow

### PR DESCRIPTION
**Pull request by wilbill**
I need to merge product branch with the master, anyone to review this

### CODE REVIEW

File: src/main/java/miu/edu/resort/controller/ProductController.java [hunk 1]
```diff
@@ -1,8 +1,63 @@
...
+import java.util.List;
+import java.util.Optional;
 
 @RestController
+@RequestMapping("api/vi/product")
 public class ProductController {
 
-}
+    @Autowired
+    private ProductServiceImpl productService;
```
Comment (karamalqinneh) [inline src/main/java/miu/edu/resort/controller/ProductController.java:20]: Create an interface then use type injection, its the best practice

File: src/main/java/miu/edu/resort/controller/ProductController.java [hunk 2]
```diff
@@ -1,8 +1,63 @@
...
+    @GetMapping
+    public ResponseEntity<?> getAllProducts() {
+        List<ProductResponse> products = productService.findAll();
+        return new ResponseEntity<>(products, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<?> getProductByID(@PathVariable Integer productId) {
+        Optional<ProductResponse> productResponseOptional = productService.findById(productId);
+        if (productResponseOptional.isPresent()) {
```
Comment (karamalqinneh) [inline src/main/java/miu/edu/resort/controller/ProductController.java:37]: This logic usually happens in the service layer. if no product is found throw an exception preferably a custom exception that should be picked up by the global exception handler

**Review by karamalqinneh — DISMISSED** [review]

**Comment by karamalqinneh** [comment]
Bro please write tests for these changes

**Comment by karamalqinneh** [comment]
@wilbill Bro the tests are required by the professor its not optional

**Review by abelgalef — APPROVED** [review]

**Pull request merged**